etc/taskcluster/decision_task.py

```diff
@@ -0,0 +1,298 @@
+        .find_or_create("build.windows_x64_dev." + CONFIG.git_sha)
+    )
+
+
+def windows_release():
+    return (
+        windows_build_task("Windows x64: release build")
+        .with_script("mach build --release",
+                     "mach package --release")
+        .with_artifacts("repo/target/release/msi/Servo.exe",
+                        "repo/target/release/msi/Servo.zip")
+        .find_or_create("build.windows_x64_release." + CONFIG.git_sha)
+    )
+
+
+def linux_wpt():
+    release_build_task = linux_release_build()
+    total_chunks = 2
+    for i in range(total_chunks):
+        this_chunk = i + 1
+        wpt_chunk(release_build_task, total_chunks, this_chunk, extra=(this_chunk == 1))
+
+
+def linux_release_build():
+    return (
+        linux_build_task("Linux x64: release build")
+        .with_script("""
+            ./mach build --release --with-debug-assertions -p servo
+            ./etc/ci/lockfile_changed.sh
+            tar -czf /target.tar.gz \
+                target/release/servo \
+                target/release/build/osmesa-src-*/output \
+                target/release/build/osmesa-src-*/out/lib/gallium
+        """)
+        .with_artifacts("/target.tar.gz")
+        .find_or_create("build.linux_x64_release." + CONFIG.git_sha)
+    )
+
+
+def wpt_chunk(release_build_task, total_chunks, this_chunk, extra):
+    name = "Linux x64: WPT chunk %s / %s" % (this_chunk, total_chunks)
+    script = """
+        ./mach test-wpt \
+            --release \
+            --processes 24 \
+            --total-chunks "$TOTAL_CHUNKS" \
+            --this-chunk "$THIS_CHUNK" \
+            --log-raw test-wpt.log \
+            --log-errorsummary wpt-errorsummary.log \
+            --always-succeed
+        ./mach filter-intermittents\
+            wpt-errorsummary.log \
+            --log-intermittents intermittents.log \
+            --log-filteredsummary filtered-wpt-errorsummary.log \
+            --tracker-api default
+    """
+    # FIXME: --reporter-api default
+    # IndexError: list index out of range
+    # File "/repo/python/servo/testing_commands.py", line 533, in filter_intermittents
+    #   pull_request = int(last_merge.split(' ')[4][1:])
+    if extra:
+        name += " + extra"
+        script += """
+            ./mach test-wpt-failure
+            ./mach test-wpt --release --binary-arg=--multiprocess --processes 24 \
+                --log-raw test-wpt-mp.log \
+                --log-errorsummary wpt-mp-errorsummary.log \
+                eventsource
+        """
+    return (
+        linux_run_task(name, release_build_task, script)
+        .with_env(TOTAL_CHUNKS=total_chunks, THIS_CHUNK=this_chunk)
+        .create()
+    )
+
+
+def linux_run_task(name, build_task, script):
+    return (
+        linux_task(name)
+        .with_dockerfile(dockerfile_path("run"))
+        .with_early_script("""
+            ./etc/taskcluster/curl-artifact.sh ${BUILD_TASK_ID} target.tar.gz | tar -xz
```

Comment:
jgraham also mentioned on IRC how a declarative syntax might be nicer for defining the task graph, but I’m not sure. For now, allowing arbitrary Python gives us a lot of flexibility. For example WPT chunking with a `for` loop rather than copy/pasting blobs of YAML.

Eventually, maybe it’ll make more sense to have a sophisticated declarative DSL (possibly based on YAML syntax) with built-in support for various things like chunking. But I think we need to go through some organic development in imperative style before we have a better idea of what this DSL should look like.